src/vs/platform/terminal/node/ptyService.ts

```diff
@@ -502,6 +502,7 @@ export class PersistentTerminalProcess extends Disposable {
 			reconnectConstants.scrollback,
 			unicodeVersion,
 			reviveBuffer,
+			this._terminalProcess.shellLaunchConfig.isFeatureTerminal || this._terminalProcess.shellLaunchConfig.hideFromUser || !this._terminalProcess.shellLaunchConfig.executable,
```

Comment:
Instead of duplicating this can we remember whether an injection happened and use that?

https://github.com/microsoft/vscode/blob/331fa9017a7d7d2370d7fc628f66bdfb61363da8/src/vs/platform/terminal/node/terminalProcess.ts#L203

There might be a nicer way but that could be a property like `TerminalProcess.shellIntegrationInjectionAttempted`. Communicating that back to the renderer to reduce this further would be ideal.

Reply:
but yes could add an event that fires just the info above

Reply:
we already have this essentially, but will also want to do this when it's disabled https://github.com/microsoft/vscode/blob/331fa9017a7d7d2370d7fc628f66bdfb61363da8/src/vs/platform/terminal/node/terminalProcess.ts#L218